pkg/infra/teams/team3/election_strat.go

```diff
@@ -85,76 +80,71 @@ func (a *AgentThree) HandleElectionBallot(baseAgent agent.BaseAgent, param *deci
 func (a *AgentThree) calcW1(state state.HiddenAgentState, w1 float64, initHP int, initStamina int) float64 {
 	currentHP := state.Hp
 	currentStamina := state.Stamina
+	// extract and normalise personality (range[0,100]), use to dictate update step size
+	personalityMod := float64(a.personality / 100)
 
 	HP := initHP - int(currentHP)
 	stamina := initStamina - int(currentStamina)
 	// alg 6
 	if stamina > 0 {
-		w1 += 0.2
+		w1 += 0.5 * personalityMod
 	} else if stamina < 0 {
-		w1 -= 0.2
+		w1 -= 0.5 * personalityMod
 	}
 	if HP > 0 {
-		w1 += 0.2
+		w1 += 0.5 * personalityMod
 	} else if HP < 0 {
-		w1 -= 0.2
-	}
-
-	if w1 > 10 {
-		w1 = 10
+		w1 -= 0.5 * personalityMod
 	}
 
-	if w1 < 0 {
-		w1 = 0
-	}
+	w1 = clamp(w1, 0, 10)
 
 	return w1
 }
 
 func (a *AgentThree) calcW2(baseAgent agent.BaseAgent, w2 float64) float64 {
 	var agentFought bool = false
 	var agentShielded bool = false
+	// extract and normalise personality (range[0,100]), use to dictate update step size
+	personalityMod := float64(a.personality / 100)
 	// iterate until we get most recent history
-	i := 0
-	itr := a.fightDecisionsHistory.Iterator()
+	i := 1
+	itr := a.fightRoundsHistory.Iterator()
 	for !itr.Done() {
 		res, _ := itr.Next()
-		i = i + 1
-
-		if i == a.fightDecisionsHistory.Len()-1 {
-			agents_attack := res.AttackingAgents()
-			agents_defended := res.ShieldingAgents()
-			itr2 := agents_attack.Iterator()
-			itr3 := agents_defended.Iterator()
-			// search for our agent in fight list
-			for !itr.Done() {
+
+		// check stats of last round from previous level (?)
+		if i == a.fightRoundsHistory.Len()-1 {
+			// search for our agent in fight list and assign action
+			agentsAttackedIDs := res.AttackingAgents()
+			itr2 := agentsAttackedIDs.Iterator()
+			for !itr2.Done() {
 				_, attackingAgentID := itr2.Next()
 				if attackingAgentID == baseAgent.ID() {
 					agentFought = true
+					break
```

Comment:
This, and `itr3` perform basically the same functionality, just with different arguments. I would package this into a function